### PR TITLE
Document YodaConditionals you must

### PR DESCRIPTION
**Pull request by Ipstenu**
A basic example of a proper (and improper) Yoda conditional, with links back to full documentation on why.

### CODE REVIEW

File: WordPress/Docs/PHP/YodaConditionsStandard.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <standard>
+    <![CDATA[
+      When doing logical comparisons involving variables, the variable must be placed on the right side. All constants, literals, and function calls must be placed on the left side. If neither side is a variable, the order is unimportant. 
+      
+For more information:
+https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="The variable $the_force is placed on the right">
```
Comment (jrfnl) [inline WordPress/Docs/PHP/YodaConditionsStandard.xml]: Please prefix this title with `Valid:`.

File: WordPress/Docs/PHP/YodaConditionsStandard.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <![CDATA[
+if ( true === $the_force ) {
+    $victorious = you_will( $be );
+}
+        ]]>
+        </code>
+        <code title="The variable $the_force has been placed on the left">
```
Comment (jrfnl) [inline WordPress/Docs/PHP/YodaConditionsStandard.xml]: Please prefix this title with `Invalid:`

File: WordPress/Docs/PHP/YodaConditionsStandard.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<documentation title="Yoda Conditional Statements">
```
Comment (jrfnl) [inline WordPress/Docs/PHP/YodaConditionsStandard.xml]: Is this a grammatically correct phrase ?

File: WordPress/Docs/PHP/YodaConditionsStandard.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    <standard>
+    <![CDATA[
+      When doing logical comparisons involving variables, the variable must be placed on the right side. All constants, literals, and function calls must be placed on the left side. If neither side is a variable, the order is unimportant. 
+
```
Comment (jrfnl) [inline WordPress/Docs/PHP/YodaConditionsStandard.xml]: Just noticed some trailing whitespace here.

File: WordPress/Docs/PHP/YodaConditionsStandard.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        </code>
+        <code title="Invalid: The variable $the_force has been placed on the left">
+        <![CDATA[
+if ( $the_force === false ) {
+    $victorious = you_will_not( $be );
+}
+        ]]>
+        </code>
+    </code_comparison>
+</documentation>
```
Comment (GaryJones) [inline WordPress/Docs/PHP/YodaConditionsStandard.xml]: New empty line needed at the end of the file.

File: WordPress/Docs/PHP/YodaConditionsStandard.xml (hunk 4 above)
Comment (GaryJones) [reply WordPress/Docs/PHP/YodaConditionsStandard.xml]: Still seems to be whitespace present here.

Screenshot 2019-06-20 at 13 30 09

**Comment by jrfnl** [comment]
Related #1722

File: WordPress/Docs/PHP/YodaConditionsStandard.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      When doing logical comparisons involving variables, the variable must be placed on the right side. All constants, literals, and function calls must be placed on the left side. If neither side is a variable, the order is unimportant.
+
+For more information:
+https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: The variable $the_force is placed on the right">
+        <![CDATA[
+if ( true === $the_force ) {
```
Comment (jrfnl) [inline WordPress/Docs/PHP/YodaConditionsStandard.xml]:
```suggestion
if ( <em>true === the_force</em> ) {
```

File: WordPress/Docs/PHP/YodaConditionsStandard.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    $victorious = you_will( $be );
+}
+        ]]>
+        </code>
+        <code title="Invalid: The variable $the_force has been placed on the left">
+        <![CDATA[
+if ( $the_force === false ) {
```
Comment (jrfnl) [inline WordPress/Docs/PHP/YodaConditionsStandard.xml]:
```suggestion
if ( <em>$the_force === false</em> ) {
```

**Comment by jrfnl** [comment]
FYI: I've opened a PR in PHPCS upstream to fix the weird word wrapping which is happening with the multi-line `standard` block: https://github.com/squizlabs/PHP_CodeSniffer/pull/2541

**Comment by jrfnl** [comment]
FYI: upstream PR for the weird line breaks has been merged and will be included in PHPCS 3.5.0.

**Comment by jrfnl** [comment]
I've rebased the PR and added one commit to make the last fixes needed. Let's get this merged!

For whomever will merge this: probably squash/merge would be a good idea.

**Review by jrfnl — APPROVED** [review]

**Review by dingo-d — APPROVED** [review]

**Pull request merged**